cpp/dolfinx/refinement/utils.h

```diff
@@ -291,11 +291,12 @@ mesh::Mesh<T> partition(const mesh::Mesh<T>& old_mesh,
   else
   {
     auto partitioner
-        = [](MPI_Comm comm, int, mesh::CellType,
-             const graph::AdjacencyList<std::int64_t>& cell_topology)
+        = [](MPI_Comm comm, int, const std::vector<mesh::CellType>& cell_types,
+             const std::vector<std::span<const std::int64_t>>& cell_topology)
     {
       const int mpi_rank = MPI::rank(comm);
-      const int num_cells = cell_topology.num_nodes();
+      int num_cell_vertices = mesh::num_cell_vertices(cell_types.front());
+      const int num_cells = cell_topology.front().size() / num_cell_vertices;
```

Comment:
Use fixed width integers to make clear that the possible range has been considered. It says a lot for developers.